### PR TITLE
Take the first step towards making the client smarter: drive it through Vue

### PR DESCRIPTION
**Pull request by ehuelsmann**
Move the main app to Vue, but hold off on the setup app (it didn't work at first, so we focused on the main app which *did* seem to work).

### CODE REVIEW

**Comment by ehuelsmann** [comment]
It's probably best to merge this as a squash-merge.

**Comment by ehuelsmann** [comment]
@ylavoie Are we missing anything documented here: https://cli.vuejs.org/guide/browser-compatibility.html#browserslist ? I don't think we have preset-env set up for `@vue/...` right now. Should we?

**Review by freelock — COMMENTED** [review]
Not really finding anything wrong with this change. Pointed out a couple tiny style things.

Overall, I haven't touched XMLHttpRequest in ages -- the fetch() API seems so much easier to use, with a simple request/options that returns a promise, no need to have event listeners. And I'm also using async/await quite a bit, which greatly facilitates debugging, makes the code even more readable.

I can add an alternative updateContent method to illustrate...

File: webpack.config.js [hunk 1]
```diff
@@ -104,12 +105,19 @@ if (TARGET !== 'readme') {
...
+
+    const vue = {
+        test: /\.vue$/,
+        loader: "vue-loader"
     };
 
     const css = {
         test: /\.css$/i,
-        use: [MiniCssExtractPlugin.loader, "css-loader"]
+        use: [ MiniCssExtractPlugin.loader, "css-loader"]
```
[the remark this answers is not in the document]
Comment (ylavoie) [reply webpack.config.js:120]: eslint + AirBNB

**Comment by ylavoie** [comment]
> @ylavoie Are we missing anything documented here: https://cli.vuejs.org/guide/browser-compatibility.html#browserslist ? I don't think we have preset-env set up for `@vue/...` right now. Should we?
> 
> Same about https://cli.vuejs.org/guide/html-and-static-assets.html#preload ?
> 
Both should be added for the actual JS & ressources.

**Comment by ylavoie** [comment]
> It's probably best to merge this as a squash-merge.

Absolutely

**Review by ylavoie — APPROVED** [review]

File: UI/js-src/lsmb/main-vue.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    },
+    beforeUpdate() {
+        domClass.remove(document.body, "done-parsing");
+    },
+    updated() {
+        domClass.add(document.body, "done-parsing");
+    }
+}).use(router);
+
+if (dojoDOM.byId("main")) {
```
[the remark this answers is not in the document]
Comment (ehuelsmann) [reply UI/js-src/lsmb/main-vue.js]: Changed.

**Pull request merged**